internal/reconstitution/controller.go

```diff
@@ -64,46 +54,58 @@ func (r *controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	logger := logr.FromContextOrDiscard(ctx).WithValues("compositionName", comp.Name, "compositionNamespace", comp.Namespace, "compositionGeneration", comp.Generation)
 	ctx = logr.NewContext(ctx, logger)
 
-	// We populate the cache with both the previous and current syntheses
-	prevReqs, err := r.populateCache(ctx, comp, comp.Status.PreviousSynthesis)
+	filled, err := r.populateCache(ctx, comp, comp.Status.PreviousSynthesis)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("processing previous state: %w", err)
 	}
-	currentReqs, err := r.populateCache(ctx, comp, comp.Status.CurrentSynthesis)
+	if filled {
```

Comment:
This reads a bit weird, does filled mean that it was filled on this tick? Or that it was already filled when observed on this tick?

Reply:
will reply in the other related comment